src/main/java/com/mongodb/jdbc/MongoSQLColumnTypeInfo.java

```diff
@@ -0,0 +1,113 @@
+package com.mongodb.jdbc;
+
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+public class MongoSQLColumnTypeInfo {
+    int jdbcType;
+    ExtendedBsonType bsonType;
+    String bsonTypeName;
+    int nullable;
+
+    MongoSQLColumnTypeInfo(MongoJsonSchema schema, int nullable) throws SQLException {
+        // All schemata except Any and AnyOf must have a bsonType.
+        if (schema.bsonType != null) {
+            this.bsonTypeName = schema.bsonType;
+            this.bsonType = MongoResultSetMetaData.getExtendedBsonTypeHelper(schema.bsonType);
+            this.jdbcType = getJDBCTypeForExtendedBsonType(this.bsonType);
+            this.nullable = nullable;
+            return;
+        }
+        if (isAny(schema)) {
+            this.bsonTypeName = "bson";
+            this.jdbcType = Types.OTHER;
+            this.bsonType = ExtendedBsonType.ANY;
+            this.nullable = ResultSetMetaData.columnNullable;
+            return;
+        }
+        // Otherwise, the schema must be an AnyOf.
+        constructFromAnyOf(schema, nullable);
+    }
+
+    // Any is represented by the empty json schema {}, so all fields
+    // will be null or false
+    private boolean isAny(MongoJsonSchema schema) {
+        return schema.bsonType == null
+                && schema.properties == null
+                && schema.anyOf == null
+                && schema.required == null
+                && schema.items == null
+                && schema.additionalProperties == false;
+    }
+
+    private void constructFromAnyOf(MongoJsonSchema schema, int nullable) throws SQLException {
```

Comment:
can we document our assumption that the schema will be simplified? specifically 1. that there will be no nested anyofs, and 2. we will not encounter an ANY in an anyof. These assumptions are baked into the code below but are not explicitly stated. as we get new team members, this may become lost knowledge.